core/src/main/java/org/testcontainers/utility/PreconditionUtils.java

```diff
@@ -0,0 +1,32 @@
+package org.testcontainers.utility;
+
+import lombok.experimental.UtilityClass;
+import org.testcontainers.containers.ContainerLaunchException;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+@UtilityClass
+public class PreconditionUtils {
+    private static final String CONTAINER_NAME_OPTION = "container_name";
+
+    public static void checkNotContainUnsupportedComposeOptions(List<File> composeFiles) {
+        composeFiles.forEach(file -> checkArgument(!containUnsupportedComposeOptions(file),
+            String.format("Compose file %s contains '%s' option which is not supported by container.",
+                file,
+                CONTAINER_NAME_OPTION)));
+    }
+
+    private static boolean containUnsupportedComposeOptions(File file) {
+        try {
+            return Files.lines(file.toPath())
+                .anyMatch(line -> line.contains(CONTAINER_NAME_OPTION));
```

Comment:
Hmm, I get that this is a simple fix - I'm just a tad wary of using `contains` to validate the compose file. Could we use Jackson to parse it?

Reply:
Yeap, i think we can! I'm trying to figure out how to add com.fasterxml.jackson.dataformat:jackson-dataformat-yaml dependency to core module

Reply:
Since Jackson uses SnakeYAML under the hood (AFAIK), I would suggest using SnakeYAML directly, jackson-dataformat-yaml is too heavy for it

Reply:
I'm afraid that snakeYAML does not support underscores in property names.
I found this declined pr https://bitbucket.org/asomov/snakeyaml/pull-requests/21/convert-snake-to-camel/diff

Reply:
this is for mapping YAML to Object, while you only need to convert it into a `Map<String, Object>`

Reply:
For services tag it will be Map<String, Map<String, Object>> because it looks like
```yaml
services:
  whoami:
    container_name: my_container
```
if it's ok i will fix it

Reply:
Be aware that:
1) there are a few versions of the format, and some old ones did not have `services:` section
2) we need to be as defensive as possible, and if the structure does not match our expectations, we should not fail, but handle it gracefully (ignore the `container_name` check)

Reply:
https://docs.docker.com/compose/compose-file/
Looks like that all versions of docker compose has `services` tag
@bsideup could you please clarify which version does not contain this tag?

Reply:
Sure!
https://github.com/testcontainers/testcontainers-java/blob/master/core/src/test/resources/compose-test.yml

Reply:
Only `v3` docker compose file format supports services, and is really only relevant to Docker Swarm mode. The current non-local-mode `DockerComposeContainer` actually only supports `v2` format right now. I've never been in a hurry to change this, as AFAICT until recently `v3` was only recommended for Swarm mode users.